docs/guides/observability/configuration.md

```diff
@@ -0,0 +1,333 @@
+
+| Category | Description | Example Events |
+|----------|-------------|----------------|
+| `observability.all` | All events (default) | All events |
+| `observability.authentication` | Authentication and authorization | Login, token validation |
+| `observability.flows` | Flow execution | Flow started, node execution |
+| `observability.authorization` | Authorization decisions | Policy evaluation |
+
+**Category Filtering Example:**
+```yaml
+observability:
+  enabled: true
+  output:
+    console:
+      enabled: true
+      categories: ["observability.authentication"]  # Only auth events
+    
+    opentelemetry:
+      enabled: true
+      categories: ["observability.flows"]          # Only flow events
+    
+    file:
+      enabled: true
+      categories: []                               # All events
+```
+
+## Common Configurations
+
+### Development Environment
+```yaml
+observability:
+  enabled: true
+  output:
+    console:
+      enabled: true
+      format: json
+      categories: ["observability.all"]
+    
+    opentelemetry:
+      enabled: true
+      exporter_type: "stdout"
+      service_name: "thunder-dev"
+      environment: "development"
+      sample_rate: 1.0
+  
+  failure_mode: "graceful"
+```
+
+### Production with Jaeger
+```yaml
+observability:
+  enabled: true
+  output:
+    console:
+      enabled: false
+    
+    opentelemetry:
+      enabled: true
+      exporter_type: "otlp"
+      otlp_endpoint: "jaeger-collector.monitoring.svc:4317"
+      service_name: "thunder-iam"
+      service_version: "1.2.0"
+      environment: "production"
+      sample_rate: 0.05                    # 5% sampling for production load
+      insecure: false
+      categories: ["observability.all"]
+    
+    file:
+      enabled: true
+      file_path: "/var/log/thunder/analytics/observability.log"
+      format: json
+      categories: ["observability.authentication", "observability.authorization"]
+  
+  failure_mode: "graceful"
+```
+
+### Staging with Tempo
+```yaml
+observability:
+  enabled: true
+  output:
+    console:
+      enabled: true
+      format: text
+      categories: ["observability.all"]
+    
+    opentelemetry:
+      enabled: true
+      exporter_type: "otlp"
+      otlp_endpoint: "tempo.staging.internal:4317"
+      service_name: "thunder-iam-staging"
+      service_version: "1.2.0-rc1"
+      environment: "staging"
+      sample_rate: 0.5                     # 50% sampling
+      insecure: true
+      categories: ["observability.all"]
+  
+  failure_mode: "graceful"
+```
+
+### Disabled (Minimal Overhead)
+```yaml
+observability:
+  enabled: false
+```
+
+## Setting Up OpenTelemetry Backends
+
+### Jaeger (Docker Compose)
+```yaml
+services:
+  jaeger:
+    image: jaegertracing/all-in-one:latest
+    ports:
+      - "16686:16686"  # Jaeger UI
+      - "4317:4317"    # OTLP gRPC
+    environment:
+      - COLLECTOR_OTLP_ENABLED=true
+```
+
+Then configure Thunder:
```

Comment:
Shall we indicate where this configuration needs to be done ? (deployment.yaml)

Reply:
This is mentioned in the begining of this document. Where should we add these commands. Here we are providing some samples. I'll add here as well

Reply:
This is fixed, add file path to all examples